tests/minifier.js

```diff
@@ -10,92 +10,309 @@ QUnit.test('`minifiy` exists', function(assert) {
   assert.ok(minify);
 });
 
+/**
+ * Converts the options in the given object to use sync callbacks where possible.
+ *
+ * @param {Object} options
+ */
+function convertOptionsSyncCallbacks(options) {
+  if (options) {
+    if (typeof options.minifyJS === 'function') {
+      var minifyJS = options.minifyJS;
+      options.minifyJS = function(text, inline, callback) {
+        callback(minifyJS(text, inline));
+      };
+    }
+    if (typeof options.minifyCSS === 'function') {
+      var minifyCSS = options.minifyCSS;
+      options.minifyCSS = function(text, type, callback) {
+        callback(minifyCSS(text, type));
+      };
+    }
+  }
+  return options;
+}
+
+/**
+ * Test the minify function synchronously, both with a callback and without.
+ *
+ * @param {Assert} assert
+ * @param {string} input
+ * @param {Object} [options]
+ * @param {string} output
+ * @param {string} [description]
+ */
+function test_minify_sync(assert, input, options, output, description) {
```

Comment:
Is it ok to switch the order of `options` and `output` so that all optional params are together?

(Same would apply to `test_minify_async`)

Reply:
diffstat is too large as it is for this file to ensure correctness as is already - I'll probably make another PR as I said back in https://github.com/kangax/html-minifier/pull/908#issuecomment-383022651

That aside, I don't understand why there is a distinction between `test_minify_sync` and `test_minify_async`:
- the use of existing API should always be synchronous
- the use of callbacks should always be asynchronous (whether some of them may complete immediately or not)

So the requirement is that all existing test cases should always go through both synchronous & asynchronous modes, the only difference should only be whether they throw/return an error or not.

Reply:
You are free to use `setTimeout()` within `minifyCSS` & `minifyJS` to emulate asynchronous behaviour - I have tried and it works on both node and web testing modes.